src/proxy/protocol.rs

```diff
@@ -35,11 +37,49 @@ impl Protocol {
             // We didn't want to keep parsing headers, just validate that
             // the first line is HTTP1.
             Ok(_) | Err(httparse::Error::TooManyHeaders) => {
+                println!("^^^^^ is H1");
                 return Some(Protocol::Http1);
             }
             _ => {}
         }
 
+        // Detect if this is a Kafka traffic
+
+        // Header:
+        //     Request/response Size => INT32
+        //     api_key => INT16
+        //     api_version => INT16
+        //     correlation_id => INT32
+        //     client_id => NULLABLE_STRING (first two bytes as the length of the client_id string)
+        const KAFKA_REQUEST_HEADER_LENGTH: usize = 4 + 2 + 2 + 4 + 2;
+        println!("length {}", bytes.len());
+        if bytes.len() >= KAFKA_REQUEST_HEADER_LENGTH {
+            let request_size =      BigEndian::read_i32(&bytes);
```

Comment:
As I mentioned above, I think we can implement this using the `bytes` crate, without adding a new dependency on `byteorder`. 

We could implement this parsing using the [`Buf`](https://docs.rs/bytes/0.4.12/bytes/trait.Buf.html) trait in `bytes`, which is implemented for `Cursor<&[u8]>`.
```rust
if bytes.len() >= KAFKA_REQUEST_HEADER_LENGTH {
    use bytes::Buf;
    use std::io::Cursor;

    let mut buf = Cursor::new(bytes);

    let request_size = buf.get_i32_be();
    let api_key = buf.get_i16_be();
    let api_version = buf.get_i16_be();
    // ... and so on
}
```
The nice thing about using `Cursor` and `Buf` is that the cursor position is advanced for you every time you read a number, so you don't have to slice the buffer from each offset. 

Alternatively, you could use the `bytes` crate's [re-export of `BigEndian`](https://docs.rs/bytes/0.4.12/bytes/enum.BigEndian.html) and this code would look identical, but we wouldn't need to add a new `extern crate` statement for `byteorder`.